lib/matplotlib/legend_handler.py

```diff
@@ -420,6 +420,30 @@ def create_artists(self, legend, orig_handle,
         return [legline]
 
 
+class HandlerPatchCollection(HandlerPatch):
+    """
+    Handler for `.PatchCollection` instances.
+    """
+    def _default_update_prop(self, legend_handle, orig_handle):
+        lw = orig_handle.get_linewidths()[0]
+        dashes = orig_handle._us_linestyles[0]
+        color = orig_handle.get_facecolor()[0]
```

Comment:
```suggestion
        facecolor = orig_handle.get_facecolor()[0]
        edgecolor = orig_handle.get_edgecolor()[0]
```
You need to handle both.

Reply:
Thanks! I will do that.